basic/2.headers/zomes/exercise/src/lib.rs

```diff
@@ -0,0 +1,36 @@
+use hdk::prelude::*;
+use holo_hash::HeaderHashB64;
+
+entry_defs![KitchenJarLabel::entry_def(), KitchenJarLabelUpdate::entry_def()];
+
+#[hdk_entry(id = "KitchenJarLabel")]
+pub struct KitchenJarLabel(String);
+
+#[hdk_entry(id = "KitchenJarLabelUpdate")]
+pub struct KitchenJarLabelUpdate{
+    label: String,
+    header_hash: HeaderHashB64,
+}
+
+#[hdk_extern]
+pub fn add_label(input: KitchenJarLabel) -> ExternResult<HeaderHashB64> {
+    let a: HeaderHash = create_entry(&input)?;
+    Ok(HeaderHashB64::from(a))
+}
+
+#[hdk_extern]
+pub fn get_label(hash: HeaderHashB64) -> ExternResult<KitchenJarLabel> {
+    let element: Element = get(HeaderHash::from(hash), GetOptions::default())?
+        .ok_or(WasmError::Guest(String::from("Could not find KitchenJarLabel")))?;
+    let option: Option<KitchenJarLabel> = element.entry().to_app_option()?;
+    let label: KitchenJarLabel = option.unwrap();
+    Ok(label)
+}
+
+#[hdk_extern]
+pub fn update_label(a:KitchenJarLabelUpdate) -> ExternResult<HeaderHashB64> {
+    let header_hash: HeaderHash = HeaderHash::from(a.header_hash);
+    let new_label: KitchenJarLabel = KitchenJarLabel(a.label);
+    let header_hash_update: HeaderHash = update_entry(header_hash, new_label)?;
```

Comment:
I wouldn't introduce updating just yet, it requires a lot of explanation and playground interactions...

Reply:
@guillemcordoba I'm not sure how well I covered explaining updates in the text: https://github.com/holochain-gym/holochain-gym.github.io/blob/headers/docs/developers/basic/headers.md
I was indeed depending on the playground to visually demonstrate the concept.

My idea was to use updates as a first step before heading to the source-chain concept.
Because you already explain the basis notion of the source-chain, but more from a CRUD view point of one entry (and its updates).
The sourcechain expands this idea to all your actions as an agent.
Do you think it is too early in the series of exercises to talk about about updates? 
If it is only the playground that is blocking, then I could look for an alternate way to demonstrate/simulate updates.
If it is conceptually too early I can split the exercise.

Let me know what you think.

Reply:
Yeah, actually updating a private entry on your source chain doesn't really make a lot of sense. I would be more comfortable splitting them up, since CRUD is actually quite tricky in holochain (specially the Update part) and I don't want to pack this exercise too much.

One more easy thing that we could do is delete a link: it's the same concept (you pass the header hash of the createlink header that created the link) but without having to go to entry CRUD. This way, you also become aware that create links actually produces a header, and that that can be interacted with.